sky/cli.py

```diff
@@ -657,6 +695,13 @@ def _launch_with_confirm(
     elif not confirm_shown:
         click.secho(f'Running task on cluster {cluster}...', fg='yellow')
 
+    if node_type is not None and idle_minutes_to_autostop is not None \
+            and maybe_status == global_user_state.ClusterStatus.UP:
+        # We do not sky.launch if interactive node is already up, so we need
+        # to update idle timeout here.
+        handle = global_user_state.get_handle_from_cluster_name(cluster)
+        backend.set_autostop(handle, idle_minutes_to_autostop)
```

Comment:
Great catch! How about we use the programmatic API `core.autostop` instead?

Also, should we make this an `elif idle_minutes_to_autostop is not None` after L718?